keps/sig-node/5593-configure-the-max-crashloopbackoff-delay/README.md

```diff
@@ -0,0 +1,1952 @@
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+At minimum, unit tests will be included confirming the backoff object is set
+properly.
+
+In this version of the proposal, there are no API schema changes or conversions
+necessary. However it is worth nothing there is one addition to an API object,
+which is a new field in the `KubeletConfiguration` Kind. Based on manual tests
+by the author, adding an unknown field to `KubeletConfiguration` is safe and the
+unknown config field is dropped before addition to the
+`kube-system/kubelet-config` object which is its final destination (for example,
+in the case of n-3 kubelets facing a configuration introduced by this KEP).
+Ultimately this is supported by the configuratinon of a given Kind's
+`fieldValidation` strategy in API machinery
+([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L584))
+which, in 1.31+, is set to "warn" by default and is only valid for API objects
+and it turns out is not explicitly set as `strict` for
+`KuberntesConfiguration` object so they ultimately bypass this
+([ref](https://github.com/kubernetes/kubectl/issues/1663#issuecomment-2392453716)).
+This is not currently tested as far as I can tell in the tests for
+`KubeletConfiguration` (in either the most likely location, in
+[validation_test](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/validation/validation_test.go),
+nor other tests in the [config
+package](https://github.com/kubernetes/kubernetes/tree/005f184ab631e52195ed6d129969ff3914d51c98/pkg/kubelet/apis/config))
+and discussions with other contributors indicate that while little in core
+kubernetes does strict parsing, it's not well tested. At minimum as part of this
+implementation a test covering this for `KubeletConfiguration` objects will be
+included in the `config.validation_test` package.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+The primary risk of this proposal is giving operators a knob which could
+potentially compromise node stability, risking the kubelet component to become
+too slow to respond and the pod lifecycle to increase in latency.
+
+All behavior changes are local to the kubelet component and its start up
+configuration, so a mix of different (or unset) max backoff durations will not
+cause issues.
```

Comment:
Just noticed that this sentence is kinda vague.

```suggestion
cause issues to running workloads.
```

Reply:
Done.